app/src/main/java/com/quran/labs/androidquran/ui/fragment/AyahTranslationFragment.java

```diff
@@ -111,34 +112,35 @@ public void refreshView() {
     final Activity activity = getActivity();
     if (activity instanceof PagerActivity) {
       PagerActivity pagerActivity = (PagerActivity) activity;
-      if (translations == null || translations.size() == 0) {
-        translations = pagerActivity.getTranslations();
-      }
 
-      if (translations == null || translations.size() == 0) {
+      allTranslationsFromDb = pagerActivity.getAllTranslationsFromDb();
+
+      if (allTranslationsFromDb == null || allTranslationsFromDb.size() == 0) {
         progressBar.setVisibility(View.GONE);
         emptyState.setVisibility(View.VISIBLE);
         translationControls.setVisibility(View.GONE);
         return;
       }
-
+      Set<String> activeTranslationsFilesNames = pagerActivity.getActiveTranslationsFilesNames();
+      if (activeTranslationsFilesNames == null) {
+        activeTranslationsFilesNames = quranSettings.getActiveTranslations();
+      }
       if (translationAdapter == null) {
-        Set<String> activeTranslations = pagerActivity.getActiveTranslations();
-        if (activeTranslations == null) {
-          activeTranslations = quranSettings.getActiveTranslations();
-        }
 
         translationAdapter = new TranslationsSpinnerAdapter(activity,
             R.layout.translation_ab_spinner_item,
-            pagerActivity.getTranslationNames(),
-            translations,
-            activeTranslations,
+            pagerActivity.getTranslationSpinnerTitles(),
+            allTranslationsFromDb,
+            activeTranslationsFilesNames,
             selectedItems -> {
               quranSettings.setActiveTranslations(selectedItems);
               refreshView();
             });
         translator.setAdapter(translationAdapter);
       }
+      else{
+        translationAdapter.updateItems(pagerActivity.getTranslationSpinnerTitles(), allTranslationsFromDb, activeTranslationsFilesNames );
```

Comment:
do we actually need this else though?

Reply:
yes we need it, if translations list order changed from `translationManagerActivity` the spinner change a accordingly